src/main/java/org/opendevstack/provision/services/JiraAdapter.java

```diff
@@ -285,10 +320,31 @@ protected int createPermissions (ProjectData project, String crowdCookieValue)
   
   protected FullJiraProject buildJiraProjectPojoFromApiProject(ProjectData s) {
     BasicUser lead = s.admins.get(0);
-    String type = "software";
+
+    String jiratemplateKeyPrefix = "jira.project.template.key.";
+    String jiratemplateTypePrefix = "jira.project.template.type.";
+    
+    String template =
+    	(s.projectType != null && 
+    		environment.containsProperty(jiratemplateKeyPrefix + s.projectType) && 
+    		projectTemplateKeyNames.contains(s.projectType)) ?
+    		environment.getProperty(jiratemplateKeyPrefix + s.projectType) : jiraTemplateKey;
+    		
+    String templateType =
+    	(s.projectType != null && 
+    		environment.containsProperty(jiratemplateTypePrefix + s.projectType) &&
+    		projectTemplateKeyNames.contains(s.projectType)) ?
+        	environment.getProperty(jiratemplateTypePrefix + s.projectType) : jiraTemplateType;
```

Comment:
Those two ternaries are difficult to read for me.

Reply:
@michaelsauter  - extracted